owncloudApp/src/main/java/com/owncloud/android/presentation/accounts/ManageAccountsDialogFragment.kt

```diff
@@ -245,26 +245,44 @@ class ManageAccountsDialogFragment : DialogFragment(), ManageAccountsAdapter.Acc
                         showErrorInSnackbar(R.string.common_error_unknown, uiResult.error)
                         Timber.e(uiResult.error)
                     }
-
                 }
             }
         }
+
+        collectLatestLifecycleFlow(manageAccountsViewModel.userQuotas) { listUserQuotas ->
+            if (listUserQuotas.isNotEmpty()) {
+                // hide the progress bar and show manage accounts dialog
+                val indeterminateProgressBar = dialogView.findViewById<ProgressBar>(R.id.indeterminate_progress_bar)
+                indeterminateProgressBar.visibility = View.GONE
+                val manageAccountsLayout = dialogView.findViewById<LinearLayout>(R.id.manage_accounts_layout)
+                manageAccountsLayout.visibility = View.VISIBLE
+
+                accountListAdapter = ManageAccountsAdapter(this)
+                accountListAdapter.submitAccountList(accountList = getAccountListItems(listUserQuotas))
+
+                val recyclerView = dialogView.findViewById<RecyclerView>(R.id.account_list_recycler_view)
+                recyclerView.adapter = accountListAdapter
+            }
+        }
     }
 
     /**
      * creates the account list items list including the add-account action in case multiaccount_support is enabled.
      *
      * @return list of account list items
      */
-    private fun getAccountListItems(): List<ManageAccountsAdapter.AccountRecyclerItem> {
+    private fun getAccountListItems(userQuotasList: List<UserQuota>): List<ManageAccountsAdapter.AccountRecyclerItem> {
         val accountList = manageAccountsViewModel.getLoggedAccounts()
         val provisionalAccountList = mutableListOf<ManageAccountsAdapter.AccountRecyclerItem>()
-        accountList.forEach {
-            provisionalAccountList.add(ManageAccountsAdapter.AccountRecyclerItem.AccountItem(it))
+        accountList.forEach { account ->
+            val userQuota = userQuotasList.firstOrNull { userQuota -> userQuota.accountName == account.name }
+            if (userQuota != null) {
+                provisionalAccountList.add(ManageAccountsAdapter.AccountRecyclerItem.AccountItem(account, userQuota))
+            }
         }
 
         // Add Create Account item at the end of account list if multi-account is enabled
-        if (resources.getBoolean(R.bool.multiaccount_support) || accountList.isEmpty()) {
+        if (resources.getBoolean(R.bool.multiaccount_support) || accountList.isEmpty() || userQuotasList.isNotEmpty()) {
```

Comment:
It was necessary in the first version because without that condition the `Add account` function was over the accounts. Now, it's not necessary so I've removed it 😄